### DISCUSSION TITLE
Timeline/Instance selector sidebar

### DISCUSSION THREAD

**Discussion by tylerknowsnothing** [hyperspacedev/hyperspace#235] [Ideas]
**Is your feature request related to a problem? Please describe.**
I find that using a menu item or hamburger menu to select which instance and/or timeline is active adds unnecessary steps to navigate Hyperspace. (Pun intended.)

**Describe the solution you'd like**
Modify the pop-out sidebar to be a persistent sidebar that shows only the icons. When swiped right, the panel would expand into the existing sidebar. In addition, break out the Instance selector so that each instance has its on icon in the minimized sidebar. 

**Describe alternatives you've considered**
An alternative approach would put tabs for the timelines at the top of the display which could be swiped to navigate. For the Instance selector, a floating button opposite the New Toot button would only appear if two or more instances have been added, and when tapped, would expand a flyout picker to select which instance is active. This is clearly a more complicated option and doesn't leverage that existing work done on that already glorious interface. 

**Additional context**
No additional context comes to mind.

**Comment by alicerunsonfedora** [comment]
I do agree that accessing timelines is a bit cumbersome if you're on a mobile device or have the window in a small size, but I don't personally agree changing the current behavior. The hamburger menu/sidebar is designed to house most of the locations as a landing point, and moving these elements around kind of defeats the purpose of it. We do have features that let you work around this, however, with the introduction of keyboard shortcuts. We have shortcuts to switch between pages so that you don't need to fiddle with the menu or sidebar at all:

- Ctrl/Command + (0, 1, 2, or 3) opens the Home, Local, and Public timelines or the Messages page, respectively
- Alt/Option + Ctrl/Command + A opens the Activity page
- Alt/Option + Ctrl/Command + N opens the Notifications page
- Alt/Option + Ctrl/Command + R opens the Recommendations page

I should note that, although this is a desktop-oriented product, Hyperspace is designed to be responsive to accommodate a wider variety of screen sizes. A persistent sidebar would make it harder for mobile users or those that have a small window to view the actual content on the screen, and I feel that this would be user-hostile. I also do not personally agree with an additional floating-action button, either. The float-action button is supposed to represent the primary action or the most common action a user will perform in that app. Having multiple buttons like this can easily confuse users, and it adds unnecessary clutter to the interface on smaller screens.

While I do agree that these changes could potentially be useful for power users or those well-versed in the fediverse, I feel that doing it this way would come at the expense of new users just getting started. Hyperspace is designed to be a simple, responsive, and fun fediverse client that new fediverse users can pick up on relatively quickly because so many other apps have a similar design language.
